src/rgw/rgw_op.cc

```diff
@@ -3634,7 +3626,32 @@ void RGWPutObj::execute()
     }
   }
 
-  op_ret = processor->prepare(store, NULL);
+  // create the object processor
+  using namespace rgw::putobj;
+  AioThrottle aio(store->ctx()->_conf->rgw_put_obj_min_window_size);
+  constexpr auto max_processor_size = std::max(sizeof(MultipartObjectProcessor),
+                                               sizeof(AtomicObjectProcessor));
+  ceph::static_ptr<ObjectProcessor, max_processor_size> processor;
```

Comment:
did we have this before?

Reply:
we had been heap-allocating the processor in select_processor(). static_ptr itself merged last year in https://github.com/ceph/ceph/pull/19079

Reply:
I remembered the latter, nice to see us using it